examples/complicated/validators.go

```diff
@@ -30,7 +30,10 @@ func (r AliasArray) Validate() error {
 		_ = i
 		_ = x
 	}
-	return &errs
+	if errs.HasErrors() {
```

Comment:
What about create some method like `ErrorOrNil` attached to `ErrList` which will do this inside:
```go
func (e *ErrList) ErrorOrNil() error {
    if e.HasErrors() {
        return e
    }
    
    return nil
}```

Name `ErrorOrNil` is just proposal, I want to discuss the idea. `pkg/errors` has very similar logic inside `wrap` method.